packages/api/src/auth/auth.controller.ts

```diff
@@ -111,6 +111,31 @@ export class AuthController {
     req.session = null;
   }
 
+  @Post("token")
+  @Header("Content-Type", "application/json")
+  @ApiOkResponse({
+    description: "Token was returned successfully",
+    schema: {
+      type: "object",
+      properties: {
+        token: { type: "string" },
+        ok: { type: "boolean" },
+      },
+    },
+  })
+  public async token(@Req() req: Request) {
+    const response = await fetch(this.configService.get("prividiumPrivateRpcUrl"), {
+      method: "POST",
+      body: JSON.stringify({
+        address: req.session.siwe.address,
+        secret: this.configService.get("prividiumPrivateRpcSecret"),
+      }),
+      headers: { "Content-Type": "application/json" },
+    });
+    const data = await response.json();
+    return data;
```

Comment:
What do you think about enforcing the response format here?

```
return {
  token: data.token,
  ok: data.ok
}
```

Another option would be parse the response and throe 424 or 500 if response doesn't follow the expected format.